fv3core/stencils/xppm.py

```diff
@@ -168,236 +106,247 @@ def xt_dxa_edge_1_base(q, dxa):
 
 
 @gtscript.function
-def xt_dxa_edge_0(q, dxa, xt_minmax):
+def xt_dxa_edge_0(q, dxa):
+    from __externals__ import xt_minmax
+
     xt = xt_dxa_edge_0_base(q, dxa)
     minq = 0.0
     maxq = 0.0
-    if xt_minmax:
+    if __INLINED(xt_minmax):
         minq = min(min(min(q[-1, 0, 0], q), q[1, 0, 0]), q[2, 0, 0])
         maxq = max(max(max(q[-1, 0, 0], q), q[1, 0, 0]), q[2, 0, 0])
         xt = min(max(xt, minq), maxq)
     return xt
 
 
 @gtscript.function
-def xt_dxa_edge_1(q, dxa, xt_minmax):
+def xt_dxa_edge_1(q, dxa):
+    from __externals__ import xt_minmax
+
     xt = xt_dxa_edge_1_base(q, dxa)
     minq = 0.0
     maxq = 0.0
-    if xt_minmax:
+    if __INLINED(xt_minmax):
         minq = min(min(min(q[-2, 0, 0], q[-1, 0, 0]), q), q[1, 0, 0])
         maxq = max(max(max(q[-2, 0, 0], q[-1, 0, 0]), q), q[1, 0, 0])
         xt = min(max(xt, minq), maxq)
     return xt
 
 
-@gtstencil()
-def west_edge_iord8plus_0(q: sd, dxa: sd, dm: sd, bl: sd, br: sd, xt_minmax: bool):
-    with computation(PARALLEL), interval(...):
-        bl = s14 * dm[-1, 0, 0] + s11 * (q[-1, 0, 0] - q)
-        xt = xt_dxa_edge_0(q, dxa, xt_minmax)
-        br = xt - q
+@gtscript.function
+def west_edge_iord8plus_0(
+    q: FloatField,
+    dxa: FloatField,
+    dm: FloatField,
+):
+    bl = yppm.s14 * dm[-1, 0, 0] + yppm.s11 * (q[-1, 0, 0] - q)
```

Comment:
I don't know if yppm is the best place for these shared constants, I just ported yppm first and then reused parts of it. These might make more sense to come from fv_tp_2d, but probably also fine to leave as-is.

Reply:
yeah I had that thought too. Perhaps we should move most of this into a new `ppm.py` file instead.